charts/descheduler/templates/cronjob.yaml

```diff
@@ -39,10 +39,12 @@ spec:
                 - "--policy-config-file"
                 - "/policy-dir/policy.yaml"
                 {{- range $key, $value := .Values.cmdOptions }}
+                  {{- if ne (kindOf $value) "invalid" }}
                 - {{ printf "--%s" $key | quote }}
-                {{- if $value }}
+                    {{- if $value }}
                 - {{ $value | quote }}
-                {{- end }}
+                    {{- end }}
```

Comment:
nit: could you fix the indentation for these?

Otherwise, looks good to me. I'll leave the lgtm for another reviewer (@ingvagabund or @seanmalloy)
/approve

Reply:
Ha. I tried finding some form of guidelines how to actually do the indentation here, but I'm very much unsure. The problem for me is that it is a nested `if`, so in a way I want to indent for that, but at the same time this `- {{ $value | quote }}` seems like it must stay exactly where it is right now. Moving the `if`s to the left gives this:

```diff
diff --git a/charts/descheduler/templates/cronjob.yaml b/charts/descheduler/templates/cronjob.yaml
index 3bdb9be5d..75ffb89e9 100644
--- a/charts/descheduler/templates/cronjob.yaml
+++ b/charts/descheduler/templates/cronjob.yaml
@@ -39,11 +39,13 @@ spec:
                 - "--policy-config-file"
                 - "/policy-dir/policy.yaml"
                 {{- range $key, $value := .Values.cmdOptions }}
+                {{- if ne (kindOf $value) "invalid" }}
                 - {{ printf "--%s" $key | quote }}
                 {{- if $value }}
                 - {{ $value | quote }}
                 {{- end }}
                 {{- end }}
+                {{- end }}
               volumeMounts:
                 - mountPath: /policy-dir
                   name: policy-volume
```

Looks off to me, WDYT?

Reply:
Ah right, you can't move those list values true. In that case I think it looks fine

Reply:
Is there a way to dry run the template renderer? Any way to unit test this?

Reply:
There is a `helm test` command. We don't currently run this as part of CI. Right now we only run `helm lint` I created #440 to track this.

Reply:
Could I suggest that IMHO it's not idiomatic to have a control statement such as `{{- if }}` nested deeper than the resulting output as it makes it harder to grep what's going on that in your example above. I'd definitely align the the `{{- range }}` and subsequent `{{- if }}` at the same level as they are part of a single logical block; but I'd possibly tab them out one level.

Reply:
Part of the issue here is probably that the control statements should start their indentation independently from the data parts. But then one would have to read the file line by line:

```yaml
              args:
                - "--policy-config-file"
                - "/policy-dir/policy.yaml"
{{- range $key, $value := .Values.cmdOptions }}
  {{- if ne (kindOf $value) "invalid" }}
                - {{ printf "--%s" $key | quote }}
    {{- if $value }}
                - {{ $value | quote }}
    {{- end }}
  {{- end }}
{{- end }}
```

This gives reasonable identation for the control flow, with the odd "yaml is unfortunately indentation-sensitive and templating it is annoying" scream. :)